wiki-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java

```diff
@@ -843,6 +846,30 @@ public boolean canModifyPagePermission(Page currentPage, String currentUser) thr
   }
   
   @Override
+  public boolean canPublicAndRetrictPage(Page currentPage, String currentUser) throws WikiException {
+    boolean canPublicAndRetrictPage = false;
+    if (currentPage.getPermissions() != null) {
+      for (PermissionEntry permissionEntry : currentPage.getPermissions()) {
+        if(permissionEntry.getId().equals(currentUser)) {
+          for(Permission permission : permissionEntry.getPermissions()) {
+            if(permission.getPermissionType().equals(PermissionType.EDITPAGE) && permission.isAllowed()) {
+              canPublicAndRetrictPage = true;

```

Comment:
If canPublicAndRetrictPage is true you should return it directly IMO. It would optimize the method since it would avoid calling getWikiByTypeAndOwner, hasAdminPagePermission, and hasAdminSpacePermission for nothing.